File: server/schemas/location.ts
import { z } from "zod";

export const locationInput = z.object({
  label: z.string().min(1),
  addressLine1: z.string().min(1),
  addressLine2: z.string().optional(),
  city: z.string().min(1),
  region: z.string().min(1),
  postalCode: z.string().min(1),
  country: z.string().min(1),
});

export type LocationInput = z.infer<typeof locationInput>;

File: src/Domain/Location.cs
namespace MeridianPlatform.Domain;

public class Location
{
    public Guid Id { get; set; }
    public required string OwnerId { get; set; }
    public required string Label { get; set; }
    public required string AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public required string City { get; set; }
    public required string State { get; set; }
    public required string PostalCode { get; set; }
    public required string Country { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
}
